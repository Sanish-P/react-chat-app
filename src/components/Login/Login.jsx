import React from 'react';
import { post } from 'axios';
import Form from '../Form/Form.jsx';

const handleLogin = (loginData) => {
  post('http://localhost:3000/login', loginData)
  .then(({data}) => {
    window.sessionStorage.setItem("access_token", data.access_token);
  })
  .catch((err) => {
    console.log(err);
  })
}

const Login = () => {
  return (
    <div className="ui middle aligned center aligned grid">
      <div className="column">
        <h2 className="ui image header">
          <div className="content">
            Log-in to your account
          </div>
        </h2>
        <Form handleLogin={handleLogin}/>
      </div>
    </div>
  )
}



export default Login;
