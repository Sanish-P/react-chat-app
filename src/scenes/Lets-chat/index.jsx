// @flow

import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import axios from 'src/utils/axios';

const LetsChat = (props) => {
  const testToken = () => {
    console.log('testToken called');
    let user = props.user_id ? props.user_id : 'no-user';
    axios
      .get(`/super-secret-resource/${user}`)
      .then(({ data }) => {
        console.log(data);
      })
      .catch(err => {
        console.log(err);
      });
  };
  console.log('calling testToken');
  testToken();
  return (
    <div>
      <h1>This is where chat happens...</h1>
    </div>
  );
};

const mapStateToProps = state => ({ user_id: state.client.user_id })

export default connect(mapStateToProps, null)(LetsChat);
