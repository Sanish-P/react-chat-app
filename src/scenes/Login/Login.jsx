// @flow

import React, { Component } from 'react';
import axios from 'src/utils/axios';
import type { RouterHistory } from 'react-router-dom';
import { bindActionCreators } from 'redux';
import {
  TransitionGroup,
  CSSTransition,
  Transition
} from 'react-transition-group';
import { connect } from 'react-redux';
import LoginForm from './components/LoginForm';
import SignUpForm from './components/SignUpForm';
import { loginRequest } from 'src/store/actionCreators/login';
import './Login.css';
import Dummy from 'src/components/Dummy';

class Landing extends Component<Props, State> {
  state = {
    newUser: true,
    items: ['hello']
  };
  checkLoggedIn = () => {
    typeof window.sessionStorage.getItem('access_token') === 'string' &&
      this.props.history.push('/lets-chat');
  };
  handleLogin = (
    { email, password }: { email: string, password: string },
    callback: Function
  ) => {
    let reqData = {
      grant_type: 'password',
      email,
      password
    };
    // axios
    //   .post('/auth/token', reqData)
    //   .then(({ data }) => {
    //     window.sessionStorage.setItem('access_token', data.access_token);
    //     window.sessionStorage.setItem('refresh_token', data.refresh_token);
    //     axios.get('/user/me').then(({ data }) => {
    //       window.sessionStorage.setItem('user_id', data.user_id);
    //       this.props.history.push('/lets-chat');
    //     });
    //   })
    //   .catch(err => {
    //     callback({ responseError: 'Email or password might be incorrect' });
    //   });
    new Promise((resolve, reject) => {
      this.props.loginRequest(reqData, resolve, reject);
    })
    .then(() => {
      this.props.history.push('/lets-chat');
    })
    .catch((e) => {
      console.log(e);
      callback({ responseError: 'Email or password might be incorrect' });
    })
  };
  handleLinkClick = (event: SyntheticMouseEvent<>) => {
    event.preventDefault();
    let newUser = this.state.newUser;
    this.setState({ newUser: !newUser });
  };
  componentWillMount() {
    this.checkLoggedIn();
  }
  render() {
    let FormComponent = null;
    let MessageComponent = null;
    if (this.state.newUser) {
      FormComponent = <SignUpForm />;
      MessageComponent = (
        <label>
          Already have an account? Login{' '}
          <a href="#" onClick={this.handleLinkClick}>
            here
          </a>
        </label>
      );
    } else {
      FormComponent = <LoginForm handleLogin={this.handleLogin} />;
      MessageComponent = (
        <label>
          Don't have an account yet? Sign Up{' '}
          <a href="#" onClick={this.handleLinkClick}>
            here
          </a>
        </label>
      );
    }
    return (
      <div className="ui center aligned three column grid">
        <div className="row">
          <Dummy />
        </div>
        <div className="column">
          <div className="ui raised segment">
            {FormComponent}
            {MessageComponent}
          </div>
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => {
  return bindActionCreators({ loginRequest }, dispatch);
};

export const Unwrapped = Landing;
export default connect(null, mapDispatchToProps)(Landing);
