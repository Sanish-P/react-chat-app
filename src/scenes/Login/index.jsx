import Login, { Unwrapped as Unmapped } from './Login';

export const Unwrapped = Unmapped;
export default Login;
