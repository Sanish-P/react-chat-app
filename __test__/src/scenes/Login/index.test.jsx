import React from 'react';
import { shallow } from 'enzyme';
import { Unwrapped as Login } from 'src/scenes/Login'

test('renders correctly', () => {
  const wrapper = shallow(<Login />);
  expect(wrapper).toMatchSnapshot();
})
