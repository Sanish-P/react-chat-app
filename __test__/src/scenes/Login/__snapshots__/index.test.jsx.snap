// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ui center aligned three column grid"
>
  <div
    className="row"
  >
    <Dummy />
  </div>
  <div
    className="column"
  >
    <div
      className="ui raised segment"
    >
      <SignUpForm />
      <label>
        Already have an account? Login
         
        <a
          href="#"
          onClick={[Function]}
        >
          here
        </a>
      </label>
    </div>
  </div>
</div>
`;
